api/pkg/s3/objectrestorepost.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/emicklei/go-restful"
+	"github.com/micro/go-log"
+	. "github.com/opensds/multi-cloud/s3/pkg/exception"
+	s3 "github.com/opensds/multi-cloud/s3/proto"
+	"golang.org/x/net/context"
+
+	//	"github.com/micro/go-micro/errors"
+	"github.com/opensds/multi-cloud/api/pkg/policy"
+)
+
+func (s *APIService) ObjectRestorePost(request *restful.Request, response *restful.Response) {
+	if !policy.Authorize(request, response, "object:restore") {
+		return
+	}
+	log.Logf("In restore POST")
+	url := request.Request.URL
+	log.Logf("URL is %v", request.Request.URL.String())
+	bucketName := request.PathParameter("bucketName")
+	objectKey := request.PathParameter("objectKey")
+	if strings.HasSuffix(url.String(), "/") {
+		objectKey = objectKey + "/"
+	}
+	restoreInput := s3.RestoreObjectInput{Key: objectKey, Bucket: bucketName}
+	ctx := context.Background()
+	objectInput := s3.GetObjectInput{Bucket: bucketName, Key: objectKey}
+	objectMD, _ := s.s3Client.GetObject(ctx, &objectInput)
+	log.Logf("The object metadata is : %v\n", objectMD)
+	if objectMD != nil {
+		client := getBackendByName(s, objectMD.Backend)
+		s3err := client.POST(&restoreInput, ctx)
```

Comment:
We need a parameter of object metadata to indicate that the object is in the restoring progress, and set that parameter before started to restore, and also set if restore succeed or failed.